system/service_windows.go

```diff
@@ -0,0 +1,88 @@
+package system
+
+import (
+	"github.com/aelsabbahy/goss/util"
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+type ServiceWindows struct {
+	service string
+}
+
+func NewServiceWindows(service string, system *System, config util.Config) Service {
+	return &ServiceWindows{service: service}
+}
+
+func (s *ServiceWindows) Service() string {
+	return s.service
+}
+
+func (s *ServiceWindows) Exists() (bool, error) {
+	m, err := mgr.Connect()
+	if err != nil {
+		return false, err
+	}
+	return s.exists(m)
+}
+
+func (s *ServiceWindows) exists(m *mgr.Mgr) (bool, error) {
+	svcs, err := m.ListServices()
+	if err != nil {
+		return false, err
+	}
+	for _, svc := range svcs {
+		if svc == s.service {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (s *ServiceWindows) Enabled() (bool, error) {
+	m, err := mgr.Connect()
+	if err != nil {
+		return false, err
+	}
+
+	// If service does not exist, it's not enabled.
+	ex, err := s.exists(m)
+	if !ex || err != nil {
+		return false, err
+	}
+
+	// Open service and check whether it's enabled (i.e. whether it starts automatically).
+	svc, err := m.OpenService(s.service)
+	if err != nil {
+		return false, err
+	}
+	cfg, err := svc.Config()
+	if err != nil {
+		return false, err
+	}
+	return cfg.StartType == mgr.StartAutomatic, nil
```

Comment:
A Windows-ism - services can have startup type
* automatic - start on boot
* automatic (delayed) - start after "a while" on boot
* manual - start on demand from system - this can be a user or another process starting the service, but won't start unless asked by _something_
* disabled - won't start on boot, won't start on-demand

Not sure how to map the semantics, TBH.

Reply:
I have updated the condition, so that `StartType != mgr.StartDisabled`. This kinda maps to the windows semantics where enabled means not disabled.

Reply:
Why have a boolean where a quadean will suffice, eh? :rekt: